index.js

```diff
@@ -61,7 +61,7 @@ Gists.prototype.list = function(options, cb) {
  */
 
 Gists.prototype.all = function(opts, cb) {
-  this.getAll('/gists/', opts, cb);
+  this.get('/gists', opts, cb);
```

Comment:
The new method in `github-base` is `.paged`. That can be used instead of `.getAll`.

Reply:
The `.pages` function is hard-coded with the `per_page=100` [link](https://github.com/jonschlinkert/github-base/blob/master/lib/utils.js#L239), which may not be desired.

Reply:
> which may not be desired.

Since the method name is `.all`, I'm thinking that it should return all gists, regardless of page size. If there is a more idiomatic page size to accomplish this we can update that in `github-base`.

Reply:
github restricts the maximum page size to 100, that's why `per_page=100` is used in the `paged` method. The `paged` method recursively requests each page of results until all the results have been fetched.

I'm not sure if there should be a limit or not on this method since when it's used to fetch all public gists, that can be a lot and the user might hit rate limits (this could happen in `github-base` too). It might be a better idea to make the `.paged` method act like a database "cursor", where it returns an object that has a `.next` method on it and the user can decide when to get the next page of results without having to track the page number or use the response headers directly. I'll open an issue on `github-base` to discuss this so it's in the correct place.